### PR TITLE
Fixes case when package-locks are without a name or a version

### PR DESCRIPTION
**Pull request by Mazurel**
This PR makes napalm work with node.js projects that do not have specified name or version in the package.json and as a result in package-lock.json. In case they are not specified, napalm will use the name or version specified as an argument for the builder, or if they are also not defined then fallbackPackageName/Version will be used.

### CODE REVIEW

File: default.nix [hunk 1]
```diff
@@ -20,10 +23,23 @@ let
...
+      # not specified by the package-lock.json
+      topPackageName =
+          if builtins.hasAttr "name" packageLock then packageLock.name
+          else if ! isNull pname then pname
+          else fallbackPackageName;
+
+      updateTopPackageVersion = obj:
+        if builtins.hasAttr "version" obj then obj
+        else obj // {
+          version = if isNull version then fallbackPackageVersion else version;
```
Comment (nmattia) [inline default.nix:40]: I might be going blind here, but where does `version` come from?

**Review by nmattia — COMMENTED** [review]
Thanks a lot! Just got one question

File: default.nix (hunk 1 above)
Comment (Mazurel) [reply default.nix:40]: `version` is a new argument for `snapshotFromPackageLockJson`. I also added `version` argument to the `buildPackage` builder and then propagated it into `snapshotFromPackageLockJson`. As a consequence, `snapshotFromPackageLockJson ` now takes two additional arguments (`pname` and `version`), but I hope it is not a problem as the only public API is `buildPackage` from what I understand.

File: default.nix [hunk 2]
```diff
@@ -20,10 +23,23 @@ let
...
+  snapshotFromPackageLockJson = packageLockJson: pname: version:
     let
       packageLock = builtins.fromJSON (builtins.readFile packageLockJson);
 
+      # Load custom name and version of the program in case it was specified and
+      # not specified by the package-lock.json
+      topPackageName =
+          if builtins.hasAttr "name" packageLock then packageLock.name
+          else if ! isNull pname then pname
+          else fallbackPackageName;
```
Comment (gytis-ivaskevicius) [inline default.nix:35]: I beleive this should work as expected:
```suggestion
      ifNotNull = a: b: if ! isNull a then a else b; # there must be something like this already in lib somewhere 
      topPackageName = packageLock.name or (ifNotNull pname fallbackPackageName);
```

File: default.nix (hunk 1 above)
Comment (gytis-ivaskevicius) [reply default.nix:40]:
```suggestion
          version = ifNotNull version fallbackPackageVersion;
```

File: default.nix [hunk 3]
```diff
@@ -20,10 +23,23 @@ let
...
+
+      updateTopPackageVersion = obj:
+        if builtins.hasAttr "version" obj then obj
+        else obj // {
+          version = if isNull version then fallbackPackageVersion else version;
+        };
```
Comment (gytis-ivaskevicius) [inline default.nix:41]:
```suggestion
      updateTopPackageVersion = obj: {
          version = ifNotNull version fallbackPackageVersion;
      } // obj;
```

**Review by gytis-ivaskevicius — APPROVED** [review]

**Comment by nmattia** [comment]
Nice, thanks a lot!

**Pull request merged**